tests/console/valgrind.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: valgrind test
+#          compile a test program and run valgrind to detect memory leaks on it
+# - compile test program with gcc
+# - Check if valgrind runs
+# - Check if valgrind correctly detects memory leaks
+# - Check if valgrind correctly detects memory leak from a forked child
+# - Check if valgrind correctly detects memory leaks from a forked child
+# - Check if valgrind correctly detects memory leaks that are still reachable
+# - Check the valgrind tool "memcheck"
+# - Check the valgrind tool "callgrind"
+# - Check the valgrind tool "cachegrind"
+# - Check the valgrind tool "helgrind"
+# - Check the valgrind tool "massif"
+# Maintainer: Felix Niederwanger <felix.niederwanger@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use registration qw(cleanup_registration register_product add_suseconnect_product get_addon_fullname remove_suseconnect_product);
+
+sub run {
+    # Preparation
+    my $self = shift;
+    $self->select_serial_terminal;
+    # development module needed for dependencies, released products are tested with sdk module
+    if (is_sle && !main_common::is_updates_tests()) {
+        cleanup_registration;
+        register_product;
+        add_suseconnect_product('sle-module-desktop-applications');
+        add_suseconnect_product(get_addon_fullname('sdk'));
+    }
+    # install requirements
+    zypper_call 'in gcc valgrind';
+    # run test script. This test script is also a standalone (copy&paste it)
+    assert_script_run 'cd /var/tmp';
+    assert_script_run 'curl -v -o /var/tmp/valgrind-test.c ' . data_url('valgrind/valgrind-test.c');
+    my $valgrind_test = <<'EOF';
```

Comment:
please put this in a bash script, and download it (same as you do in l47)

Reply:
I wrote to @grisu48 that he can use also this way, I don't mind to use proposed download,
but benefit of this way is that you can see the script with the test itself.
Don't need to dig for it when debugging.

Reply:
I'm still on SP12-SP3 and lower, it will fail there. I'm working on this at the moment.

Reply:
SLE12-SP3 and lower will fail, because of the missing SDK. Runs pass for SLE15-SP1, SLE15, SLE12-SP4+5 and openSUSE Leap and Tumbleweed. It's good to go from my side